File: leetCode/fukun/101-200/lc_118.js
const generate = function (rows) {
    return rows === 0 ? [] : (() => {
        const op = generate(rows - 1);
        op.push([1]);
        for (let i = 0; i < rows - 1; i++) op[rows - 1].push(op[rows - 2][i] + (i === rows - 2 ? 0 : op[rows - 2][i + 1]));
        return op;
    })();
};

console.log(generate(10));

File: leetCode/XSS/lc_118.java
import java.util.ArrayList;
import java.util.List;

public class lc_118 {
    public static void main(String[] args) {
        //杨辉三角
    }
    public static class TreeNode {
        int val;
        TreeNode leftp;
        TreeNode rightp;
        //TreeNode parent;
        TreeNode next;
    }
    static class Solution {
        public List<List<Integer>> generate(int numRows) {
            List<Integer> pre=new ArrayList<>();
            List<Integer> now=new ArrayList<>();
            List<List<Integer>> res=new ArrayList<>();
            pre.add(1);
            res.add(pre);
            //pre=new ArrayList<>();
            for(int i=2;i<=numRows;i++)
            {
                for(int j=0;j<i;j++)
                {
                    int val=0;
                    if(j-1>=0)val+=pre.get(j-1);
                    if(j<pre.size())val+=pre.get(j);
                    now.add(val);
                }
                pre=now;
                res.add(now);
                now=new ArrayList<>();
            }
            return res;
        }
    }
}
